pkg/dp-server/server/server.go

```diff
@@ -80,11 +80,21 @@ func (d *DpServer) Start(stop <-chan struct{}) error {
 		return err
 	}
 	server := &http.Server{
-		ReadHeaderTimeout: time.Second,
+		// Providing ReadHeaderTimeout value here just to satisfy gosec.
+		// It's configurable and overwritten few lines below. Pay attention that
+		// the default value (replicated in this line) is set to 5s, and not 1s
+		// like in the other servers. In rare cases, when Kuma CP was restarting,
+		// 1s was insufficient and DPs were failing to reconnect (we observed
+		// this in Projected Service Account Tokens e2e tests, which started
+		// flaking a lot after explicitly setting this value to 1s)
+		ReadHeaderTimeout: 5 * time.Second,
```

Comment:
Can't we have a util with mustReadDuration? Do we do something like this for other Timeouts?

Shouldn't this comment be on the configuration instead of hidden here in the code?

Reply:
Actually we have a lot of examples of duration duration config right. For example: https://github.com/kumahq/kuma/blob/0e74905252e4995a24737ee5ee002adfcdb348b5/pkg/config/dp-server/config.go#L256-L259

Reply:
> Shouldn't this comment be on the configuration instead of hidden here in the code?

It is commented in both places

Reply:
> Actually we have a lot of examples of duration duration config right. For example:
> 
> https://github.com/kumahq/kuma/blob/0e74905252e4995a24737ee5ee002adfcdb348b5/pkg/config/dp-server/config.go#L256-L259

ack

Reply:
addressed